helm/cluster/templates/clusterapi/controlplane/_helpers_flatcar.tpl

```diff
@@ -20,6 +20,18 @@ containerLinuxConfig:
 {{- if (((((($.Values.providerIntegration.controlPlane).kubeadmConfig).ignition).containerLinuxConfig).additionalConfig).systemd).units }}
 {{- include "cluster.internal.kubeadm.ignition.containerLinuxConfig.additionalConfig.systemd.units" $.Values.providerIntegration.controlPlane.kubeadmConfig.ignition.containerLinuxConfig.additionalConfig.systemd.units }}
 {{- end }}
+- name: setup-apiserver-environment.service
```

Comment:
Let's put this unit to a new helper named `"cluster.internal.controlPlane.kubeadm.ignition.containerLinuxConfig.additionalConfig.systemd.units.default"` (in this same file) so it's done similarly to other cases.

Then it would be included above under:

```
include "cluster.internal.kubeadm.ignition.containerLinuxConfig.additionalConfig.systemd.units.default"
```

So we get (the one in the middle would be new):
```
containerLinuxConfig:
  additionalConfig: |
    systemd:
      units:
      {{- include "cluster.internal.kubeadm.ignition.containerLinuxConfig.additionalConfig.systemd.units.default" $ | indent 6 }}
      {{- include "cluster.internal.controlPlane.kubeadm.ignition.containerLinuxConfig.additionalConfig.systemd.units.default" $ | indent 6 }}
      {{- include "cluster.internal.controlPlane.kubeadm.ignition.containerLinuxConfig.additionalConfig.systemd.units" $ | indent 6 }}
```

This way we always render the systemd units in the following order:
1. default units for both control plane and worker nodes
2. default units for control plane nodes
3. provider-specific systemd units

So we get predictable and consistent resources across providers as well.

We have this similar ordering in many other places in the chart, the intent is still not documented, will try to fix that as much as possible.

Reply:
P.S. Template names are too long I think, I'll open a PR some time next week to simplify them a bit.

e.g. change 

from

```
cluster.internal.controlPlane.kubeadm.ignition.containerLinuxConfig.additionalConfig.systemd.units
```

to

```
cluster.internal.controlPlane.ignition.systemd.units
```

or 

```
cluster.internal.controlPlane.os.systemd.units
```

or even shorter

```
cluster.internal.controlPlane.systemd.units
```

Reply:
Makes a lot of sense, thanks!
I've moved it in the last commit :)